src/backend/distributed/relay/relay_event_utility.c

```diff
@@ -637,6 +637,13 @@ AppendShardIdToName(char **name, uint64 shardId)
 	uint32 longNameHash = 0;
 	int multiByteClipLength = 0;
 
+	if (nameLength >= NAMEDATALEN)
```

Comment:
I suppose that Citus code might call this function with a *name longer than NAMEDATALEN - 1, so adding this check provides some benefit.